docs: New user tutorial #221

klfoulk16 opened the pull request:
docs: Create outline for new user tutorial.

klfoulk16 commented:
I've been trying to tidy up my commit history as it involves lots of commits for typo fixes and such...this has caused me to have to force push my branch. Is this the right way to do this? I feel like I'm doing something wrong here.

pilgrim2308 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,162 @@
+
+![Screenshot Showing Help](getting-started-imgs/help.png)
+
+Let's try out a few of these shortcuts out!
```
Let's try out a few of these shortcuts ~~out~~!

pilgrim2308 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,162 @@
+
++ Second, type <kbd>#</kbd> to view all messages you were mentioned (@'ed) in.
+
++ Third, type <kbd>/</kbd> to jump to the search bar at the top of the in the middle column. Type <kbd>esc</kbd> to move your cursor out of the search box so we can use our keyboard shortcuts again (otherwise you'd just be typing them in the search box!).
```
Accidentally missed a word: `search bar at the top of the` screen `in the middle column`

pilgrim2308 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,162 @@
+
+# Narrowing
+
+You'll be hearing the term 'narrow' a lot in in Zulip. What does it mean?
```
'narrow' a lot in ~~in~~ Zulip

pilgrim2308 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,162 @@
+
+## Narrow to a Topic
+
+First, let's move our cursor back to the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream in the left column - otherwise the following keyboard shortcut won't work. With our cursor on the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream, type <kbd>t</kbd> to toggle on a list of topics within that stream. Move your cursor to a random topic and then pre ...
```
A small typo in the last sentence-  
`your current narrow is dispalyed` --> `your current narrow is displayed`

pilgrim2308 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,162 @@
+3. Press <kbd>ctrl</kbd><kbd>f</kbd> and a list of potential recipients will appear in the footer (highlighted in red). 
+4. Press <kbd>ctrl</kbd><kbd>f</kbd> until the name of the person you want to send a message to is highlighted (IMG 2). 
+5. Press <kbd>tab</kbd> to jump to the message section and type in your message. 
+6. Type <kbd>ctrl</kbd><kbd>d</kbd> to send
```
~~Type~~ Press

pilgrim2308 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,162 @@
+
+# Edit a Message
+
+Let's try editing the new topic we added to the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream. To edit a message, first make sure your cursor is resting on the message you want to edit then type <kbd>e</kbd>. Change whatever you'd like to change, then type <kbd>ctrl</kbd><kbd>d</kbd> to send the updated the message. 
```
to send the updated ~~the~~ message.

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,162 @@
+To send a PM:
+1. Type <kbd>x</kbd> from anywhere in Zulip Terminal. A message editor will pop open at the bottom of the middle column. 
+2. Type in part of the name of the person you'd like to send a PM to (IMG 1)
+3. Press <kbd>ctrl</kbd><kbd>f</kbd> and a list of potential recipients will appear in the footer (highlighted in red). 
```
> Press <kbd>ctrl</kbd><kbd>f</kbd> and a list of potential recipients will appear in the footer (highlighted in red).

I ran into an issue where this list didn't show up; the header just stayed like this:

![image](https://user-images.githubusercontent.com/22248748/109050885-ea73b680-76a7-11eb-88ca-25e627c33467.png)

But then I saw a recent [discussion](https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/PM.20compose.20bug.20%28bad.2Fno.20recipients%29.20.23T935/near/1123180) on CZO and thought I'd try grabbing the app from current `main` (instead of the latest released version) to see if the issue was fixed…

…and it was! 🎉 So, no problem here.

chrisbobbe replied:
Also, today I learned that you can use `<kbd>x</kbd>` and it'll render something that looks like a keyboard key: <kbd>x</kbd>. Neat, thanks for that!

neiljp replied:
Do you think it's OK to have the tutorial be consistent with an unreleased version, then?

chrisbobbe replied:
> Do you think it's OK to have the tutorial be consistent with an unreleased version, then?

Hmm, good question. For purely developer-facing documentation (which is most or all of zulip-mobile's documentation, I guess), I wouldn't hesitate, because it's a safe assumption that developers are working on the latest `main` or at least can get there pretty easily (I used the [`Installation -> Latest` part of the README](https://github.com/zulip/zulip-terminal#installation)).

zulip/zulip.git keeps plenty of [user-facing documentation](https://github.com/zulip/zulip/tree/master/templates/zerver/help), so I looked there for some clues. I see a few commits (zulip/zulip@160cc5120, zulip/zulip@1d5aa2e51) where some functionality changes, and the user-facing doc is updated simultaneously. So, for some time, the user-facing doc on GitHub is inconsistent with any released version.

…but this turns out to not be a huge problem for that project, because the most natural way for a user to access the doc isn't through GitHub, but through the Zulip installation itself, where, if all goes to plan, the documentation will be precisely aligned with everything else. E.g., for CZO, you'd go to https://chat.zulip.org/help/; or, for the Recurse Center, https://recurse.zulipchat.com/help/.

In ZT's case, GitHub is the place a user goes for user-facing documentation like this tutorial, and GitHub naturally shows the latest code, which is often unreleased. Hmm.

I guess I don't have a definite answer. I suppose one solution would be to point users to a published website for user-facing documentation, and have it default to the latest stable release while allowing them to select something else. Like [React Native](https://reactnative.dev/docs/getting-started):

<img width=400 src="https://user-images.githubusercontent.com/22248748/109060834-e51c6900-76b3-11eb-9948-979799cd2f3d.png" />

But that seems like a lot of effort.

chrisbobbe replied:
Also, I guess, the more frequently releases get made, the less one has to think about an inconsistency like this being a problem.

chrisbobbe reviewed (commented):
Thanks, @klfoulk16! I've just used this guide to get acquainted with zulip-term, and everything worked!

@pilgrim2308 has already mentioned the few typos I noticed.

Also, I'd second @neiljp's [idea](https://chat.zulip.org/#narrow/stream/19-documentation/topic/ZT.20Getting.20Started.20Tutorial/near/1123183) to squash all these commits into one; it's probably what I'd do in zulip-mobile. It's sensible to use separate commits while working on it—I see that they represent progressively complete drafts that all lead to the final draft—but I expect future developers reading this file's history will be much more interested in the work as a whole, than in the progress marked in each non-final draft. 🙂 What do you think?

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,162 @@
-Hi, are you new to [Zulip](https://github.com/zulip/zulip)? If so, we recommend trying out our [web-client](https://chat.zulip.org) first to understand the concept of [streams/topics/PMs](https://zulip.com/help/about-streams-and-topics) in the world of Zulip. Just sending a message or two in [#test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream should help you get started.
+Hi, are you new to [Zulip](https://github.com/zulip/zulip)? If so, I recommend trying out our [web-client](https://chat.zulip.org) first to understand the concept of [streams/topics/PMs](https://zulip.com/help/about-streams-and-topics) in the world of Zulip. Just sending a message or two in [#test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream should help.
```
Ah, I almost forgot—along with the link to CZO (chat.zulip.org), I think it'd be good to link to the [CZO community doc](https://zulip.readthedocs.io/en/latest/contributing/chat-zulip-org.html), which has lots of useful information that we'd like people to be aware of.

klfoulk16 replied:
Added the plug in the first paragraph

klfoulk16 commented:
Thanks for all the comments! I went through and fixed everything. As for the autocomplete...I left it in the tutorial but I made clear that you need to be using the version from Github (as opposed to the stable release). I also added more information about how to send PMs other ways.

neiljp commented on docs/getting-started.md:
```diff
@@ -1,21 +1,168 @@
+
+## Create a New Topic
+
+To create a new topic in a stream, type <kbd>c</kbd> and a message editor will pop open at the bottom of the middle column. Make sure your cursor is in the middle column, otherwise nothing will happen when you type <kbd>c</kbd>.
```
Not pertinent to the tutorial right now, but do you think we should support this working in the stream or topic list?

klfoulk16 replied:
I do feel like it's a bit confusing how it works in the middle column but nowhere else. Considering how the c key isn't repurposed for something else in other sections, I think it would be nice if the c key was consistent across ZT

neiljp commented on docs/getting-started.md:
```diff
@@ -1,21 +1,168 @@
+
+# Edit a Message
+
+Let's try editing the new topic we added to the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream. To edit a message, first make sure your cursor is resting on the message you want to edit then type <kbd>e</kbd>. Change whatever you'd like to change, then type <kbd>ctrl</kbd><kbd>d</kbd> to send the updated message. 
```
It may be worth noting about time limits for editing, and content vs topic editing? Or will that be too complex?

klfoulk16 replied:
Added both of those

neiljp reviewed (commented):
@klfoulk16 I've left quite a few points, thought these are not all directly applicable to the document/PR itself. I think this will really help people get started :+1:

klfoulk16 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,168 @@
+
+### Thumbs-up Reaction
+
+If you want to show you agree with the current message, type <kbd>+</kbd> to add a thumbs-up reaction. Type <kbd>+</kbd> again to remove the thumbs-up.
```
I added this but I think it overcomplicates things...

klfoulk16 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,168 @@
+
+![Screenshot Showing Thumbs-up](getting-started-imgs/thumbs-up.png)
+
+## Send a PM
```
Changed to "Reply via a Private Message"

klfoulk16 commented:
I have updated the PR according to your recommendations and resized the images. I decided to go with a 75% width that way the images will dynamically resize to the browser window (I couldn't decide on an exact size that worked well for all).

neiljp commented on docs/getting-started.md:
```diff
@@ -1,21 +1,180 @@
+
+Let's try sending a PM to the author of a message. Select the message you sent to the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream earlier and press <kbd>R</kbd> to send a PM to yourself. Type your message in the message editor that appears at the bottom of the middle column and then type <kbd>ctrl</kbd><kbd>d</kbd> to send. Press the <kbd>P</kbd> hotkey as we did earli ...
+
+#### Send a Private Message to Someone New
```
Should this be a `###`-level title?

neiljp commented on docs/getting-started.md:
```diff
@@ -1,21 +1,168 @@
+
+You can also send a PM by moving your cursor to the list of "Users" in the left column and selecting the name of the person you'd like to send a message to.
+
+### Send a PM with the New Autocomplete Feature
```
You changed this now, but yes I think I meant those two sections felt like one section.

I think we should handle the tutorial/release synchronization issue separately.
Is your last comment regarding my response elsewhere in the PR about using the tutorial source at run-time?

klfoulk16 replied:
Yup my last commend was in regards to your response elsewhere in the PR about using the tutorial source at run-time. I'll link to the FAQ at the end. I think for now the section about sending a PM can stay in this tutorial, I think it flows well.

klfoulk16 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,180 @@
++ **Noun**: A narrow is a set of filters for Zulip messages, that can be based on many different factors (like sender, stream, topic, search keywords, etc.).
++ **Verb**: The process of navigating to a different narrow. For example, to go from viewing a stream to a topic within that stream.
+
+There are several different ways to 'narrow to a narrow' so to speak. For example, you can use the keyboard shortcuts as we did the section above. When you used the keyboard shortcut <kbd>P</kbd> to view your Private Messages (PMs), you narrowed to the 'Private Messages' narrow. Let's learn about some other ways to narrow.
```
I moved this section into the "Verb" section and while it makes it a bit verbose, I think the flow is restored and it works well.

klfoulk16 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,180 @@
 
-#### What do you see?
-![Screenshot](https://user-images.githubusercontent.com/9568999/106061037-b659a580-60a9-11eb-8ff8-ea1c54ac9084.png)
+This tutorial was designed to be interactive. We highly recomend opening up Zulip Terminal and following along! We'll try out a few keyboard shortcuts and send a couple messages to help familiarize ourselves with Zulip Terminal. Here's an overview of the features we'll be walking you through:
```
I added this to the "Open App" section

klfoulk16 commented:
I decided I should remove the extra information about reactions and keep it short and sweet (the extra text was fairly verbose and intuitive)...let me know if you disagree and I'll add it back:

Reactions for a message will be displayed right below it, such as the "thumbs-up" reaction in the screenshot below. The number beside a reaction is the number of people who sent that reaction (currently it's 1). The fact that the reaction is highlighted pink means that we sent that reaction. The colors of reactions sent by others will be inverted (notice how the text is pink and the highlight is none for the reactions "octopus" and "+1" a few messages down).

neiljp commented on docs/getting-started.md:
```diff
@@ -1,21 +1,180 @@
+
+## Edit a Message
+
+Let's try editing the new topic we added to the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream. By default, Zulip allows you to edit the content of your messages within 10 minutes of when you send them.
```
This 10m default is good, but if on chat.zulip.org (or other servers) these values may be different, including for topics - eg. it's currently 1h on chat.zulip.org for content. It may be too complex, but we could link to https://chat.zulip.org/help/configure-message-editing-and-deletion .

klfoulk16 replied:
I think the article is a bit out of the league of this tutorial given that the feature is only accessible by administrators. However I did add some clarification about the editing time.

neiljp commented on docs/getting-started.md:
```diff
@@ -1,21 +1,180 @@
+
+To edit a message, first make sure your cursor is resting on the message you want to edit then type <kbd>e</kbd>. You can edit the message's content, topic or stream. By default your cursor will be in the content portion of the message, press <kbd>tab</kbd> to move it to the stream or topic portion if you'd like to edit those as well. Change whatever you'd like to change, then type <kbd>ctrl</kbd ...
+
+To delete a message, delete all of the message's text.
```
...before updating with ctrl d?

neiljp commented on docs/getting-started.md:
```diff
@@ -1,21 +1,180 @@
+
+Let's try editing the new topic we added to the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream. By default, Zulip allows you to edit the content of your messages within 10 minutes of when you send them.
+
+To edit a message, first make sure your cursor is resting on the message you want to edit then type <kbd>e</kbd>. You can edit the message's content, topic or stream. By default your cursor will be in the content portion of the message, press <kbd>tab</kbd> to move it to the stream or topic portion if you'd like to edit those as well. Change whatever you'd like to change, then type <kbd>ctrl</kbd ...
```
It may be worth noting the equivalence to the sending key, or that may be confusing?

We could briefly mention that this (and deleting) normally leave the previous edition(s) available (with a potential link to "extra things to explore - message info -> edit history"?)

klfoulk16 replied:
Interesting, I had no idea there was a way to view previous versions of a message. I see a tutorial for how to do this on Zulip, but is it possible to do this on ZT? Right now I've linked to the Zulip tutorial about it. I think explaining how is outside the scope of this tutorial

neiljp replied:
Yes, we support this right now with `i` on a message, and if there is history you can show that using `e`.

neiljp reviewed (commented):
@klfoulk16 This is looking really great :+1:

We could iterate on variations to the text, but this seems very close to ready. Some of the things that stand out for me are:
* Making the images more polished - mostly as we discussed. The smaller images already help, and when the text in them is larger (ie. fewer rows/columns) then that will too. I mentioned whether you want to use a demo/bot user for the screenshots in the stream. Also I wasn't sure if you had particular refinements planned regarding the annotations - do you have multilayer images which you're converting to the final versions, to explore different annotations? I don't think we need particularly polished versions initially, particularly if we have the originals to work with.
* Adding some emphasis on signing up for an account on chat.zulip.org (or some other server) early on "Ensure you have an account / sign up", to ensure the "learning by doing" can take place, given that users can't sign up using zulip-terminal yet.

Other than that, you've resolved some comments - but I'm not sure if the related version is pushed to the PR branch yet? eg. regarding one title level, the verb/noun splitting, etc.

I'd like to get the next iteration out for external review again, as it'd be good to hear from new users' :)

neiljp commented:
@klfoulk16 We're discussing the images still, but I think we're good to get more feedback?

@chrisbobbe Would you be up for taking another look, if you haven't been following along?

Follow-up work that comes to mind: (please chip in with other ideas if you have them!)
* Add FAQ entries like "Can I see the past versions (history) of a message?", "Could you explain what reactions are and how they're displayed" (like the extra text above that was cut), and various similar aspects, so we can show more bullet points at the end of the tutorial.
* Simplify where we can use `c` (related to #871), allowing the tutorial to be simplified :)
* Linking to the tutorial doc at runtime (we'd have to figure out how to render it, unless we just link to a github page)
* Rather beyond the direct scope of ZT, but whether we can store screenshots as colored/positioned text, maybe via asciinema, to allow integration into the application, presentations, ... (generalizing to add captions)

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+
+### Narrow to a Topic
+
+First, let's move our cursor back to the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream in the left column - otherwise the following keyboard shortcut won't work. With our cursor on the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream, type <kbd>t</kbd> to toggle on a list of topics within that stream. Move your cursor to a random topic and then pre ...
```
> "test here: topic you chose"

nit: Is it possible to use the specific right-arrow character here instead of a colon? 
![image](https://user-images.githubusercontent.com/22248748/110875935-90c4dc00-82a4-11eb-97f7-2e015bb21aa0.png)

klfoulk16 replied:
good idea!!

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+
+## Reading Messages
+
+Time to read some messages! Read messages by scrolling up (<kbd>Up</kbd>/<kbd>k</kbd>) and down (<kbd>Down</kbd>/<kbd>j</kbd>). Any message that your cursor selects will be marked as read. If a message was visible, but never selected, it will not be marked as read. Messages marked as 'read' are colored white and unread messages are blue - notice how in the image below all of the messages below my ...
```
nit: "unread messags"

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+
+Let's try out a few of these shortcuts!
+
++ First, type <kbd>P</kbd> to view your Private Messages (PMs).
```
There may be several other places where this applies; I see at least <kbd>S</kbd> is mentioned later in the doc.

neiljp replied:
We don't currently standardize on this, though it has been brought up previously. This could probably be added as a point in #945, ie. to translate upper(x) to shift+lower(x).

klfoulk16 replied:
I changed all of the instances in this document

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+
+You can send a PM by moving your cursor to the list of "Users" in the left column and selecting the name of the person you'd like to send a message to.
+
+If you grabbed the latest unreleased version from [Github](https://github.com/zulip/zulip-terminal) or are running a version greater than 0.6.0 (use `zulip-term --version` to check), you can try out our new autocomplete feature! 
```
nit: "Github" should read "GitHub"

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+If you grabbed the latest unreleased version from [Github](https://github.com/zulip/zulip-terminal) or are running a version greater than 0.6.0 (use `zulip-term --version` to check), you can try out our new autocomplete feature! 
+
+To send a PM using the autocomplete feature:
+1. Type <kbd>x</kbd> from anywhere in Zulip Terminal. A message editor will pop open at the bottom of the middle column. 
```
> Type <kbd>x</kbd> from anywhere in Zulip Terminal.

_Almost_ anywhere, right—it won't work if you're focused on the compose box, or the search box, etc. Could just say "use the <kbd>x</kbd> hotkey"; that'd be correct.

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+
+Try following the steps above to send a message to the Welcome Bot!
+
+*You must use <kbd>ctrl</kbd><kbd>f</kbd> to autocomplete the name and zulip address of the person you want to PM. If you just type in their full name Zulip Terminal will throw an error.*
```
> zulip address

nit: capitalize Zulip

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+
+To create a new topic in a stream, type <kbd>c</kbd> and a message editor will pop open at the bottom of the middle column. Make sure your cursor is in the middle column, otherwise nothing will happen when you type <kbd>c</kbd>.
+
+By default this will send a new message to the stream of whichever message you have selected. Let's send a test message to [# test here](https://chat.zulip.org/#narrow/stream/7-test-here). If you're not already in the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream, replace the stream name with 'test here'. Use <kbd>tab</kbd> to jump to the topic section and type in a topi ...
```
> [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) was for things like this tutorial

nit: "was for things" should read "is for things", or maybe "was made for things"

chrisbobbe commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+
+Let's try editing the new topic we added to the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream. By default, Zulip allows you to edit the content of your messages within 10 minutes of when you send them, but some servers allow you to edit messages for longer. You can edit message content for up to one hour after sending on the Zulip Community server.
+
+To edit a message, first make sure your cursor is resting on the message you want to edit then type <kbd>e</kbd>. You can edit the message's content, topic or stream. By default your cursor will be in the content portion of the message, press <kbd>tab</kbd> to move it to the stream or topic portion if you'd like to edit those as well. Change whatever you'd like to change, then press <kbd>ctrl</kb ...
```
> You can edit the message's content, topic or stream.

Huh, interesting; @neiljp, when it says you can edit the stream of a message, is this the relatively new "move topics between streams" feature? Hmm, but IIUC #826 suggests that that feature isn't yet fully handled. So I guess I'm not totally sure what's meant to happen if you edit the stream. 🙂

As one data point, I tried editing the stream of a message in `#test here`, so its stream would instead be a different one, a private one I had made months ago for some testing.

I confirmed in the web app that the message remained in `#test here`, and no new message appeared in the private stream. The message's edit history in the web app looks like this:

<img width=500 src="https://user-images.githubusercontent.com/22248748/110877664-bf908180-82a7-11eb-99d2-057e1f2ba455.png" />

chrisbobbe reviewed (commented):
Thanks, @klfoulk16!

> Would you be up for taking another look, if you haven't been following along?

Sure!

I've just gone through it again, and overall it still looks great! I've added a few small comments below (sorry if some of the points already have discussions that I missed 🙂).

klfoulk16 commented on docs/getting-started.md:
```diff
@@ -1,21 +1,186 @@
+
+If you have issues opening Zulip Terminal, check out our [Running for the First Time](https://github.com/zulip/zulip-terminal#running-for-the-first-time) section.
+
+*Note: This tutorial assumes you're on the Zulip Community Server. While you can still follow along if you're not, various examples won't be the same (such as the [# test here](https://chat.zulip.org/#narrow/stream/7-test-here) stream).*
```
I guess I always felt like there was a space there (kinda looks like it in the desktop app and ZT). I think the person who originally started this tutorial had a space there so I continued that syntax

klfoulk16 commented:
Ready for review!

neiljp commented on docs/getting-started.md:
```diff
@@ -1,21 +1,194 @@
+
+## What's Next?
+
+That's all for today! Now that you're acquainted with Zulip Terminal, how about stopping by [our stream](https://chat.zulip.org/#narrow/stream/206-zulip-terminal) and saying hi? We'd love to hear how things went!
```
My only thought here is that after all the tutorial, it would be good to point out that if the user has a community server account they can subscribe to #**zulip-terminal** in the webapp and use zulip-terminal to chat in that stream :)

I'm not sure if there's a direct link to go to a subscription management; we don't have subscribing to streams in ZT yet.

neiljp reviewed (commented):
@klfoulk16 This looks great! I've left a few points, but really they're just polish which we could add later.

We can move the developer notes at the bottom into a markdown comment right at the top of the the document?

We didn't get a conclusion on the space in '# test here' though?

I'd be happy to hear from @chrisbobbe again, or others, or look to merge :)

klfoulk16 commented:
Tackled all your comments and fixed the odd # test here space. I think it's ready!

neiljp commented:
@klfoulk16 Thanks again for all your work on this! I pushed a slightly updated version temporarily to https://github.com/neiljp/zulip-terminal/blob/review-original-922/docs/getting-started.md - you can check the latest commit there for the few changes I made on top, but other than some minor textual changes, the biggest net effect was using `pngcrush` which losslessly reduces image size by ~75%!

This is in my merge queue based on that version, so let me know if I've missed anything or you have any last-minute thoughts :)

klfoulk16 commented:
I just read through it and everything sounds/looks great! That's awesome you were able to compress the images that much! I should have thought of that.

neiljp commented:
@klfoulk16 Thanks again! Merged :tada:
